katz/src/main/kotlin/katz/data/Kleisli.kt

```diff
@@ -0,0 +1,18 @@
+package katz.data
+
+import katz.Functor
+import katz.HK
+import katz.Monad
+
+class Kleisli<F, D, out A>(val run: (D) -> HK<F, A>) {
```

Comment:
I have just removed `Reader`, added the `Reader alias to `ReaderT<Id, D, A>`, and added more combinators like ask, pure and flatten.

I am facing a problem with a simple extension function that we had for the `Reader` to lift it from any `(A) -> B` function. I need to lift a method like `d -> HKT<F,A>` from the function to return a proper reader instance using the alias, but I have not been able to do it yet. Some help on that one please ?